workers/src/main/java/edu/cornell/testenv/testrunner/JUnit5TestRunner.java

```diff
@@ -21,48 +30,45 @@ public class JUnit5TestRunner implements TestRunner {
      * @return boolean of whether all JUnit tests passed
      */
     @Override
-    public boolean runTest(TestEnvContext<String> context) {
+    public boolean runTest(TestEnvContext<String> context, TestOutputStream outputStream, File rootDir) {
         try {
+            //TODO: Assumption is class files live in /build/classes/java/ likely needing a resolve
+            String baseRootPath = rootDir.toString() + "/build/classes/java/".replaceAll("/", File.separator);
 
-            List<String> classPaths = context.getTestClasses();
-
-            // Setup Launcher to find and builder test map for JUnit
-            LauncherDiscoveryRequest request = LauncherDiscoveryRequestBuilder.request()
-                    .selectors(classPaths.stream().map(
-                            path -> DiscoverySelectors.selectClass(path)
-                    ).toList()).build();
-
-            // Test Result Listener
-            SummaryGeneratingListener listener = new SummaryGeneratingListener();
-
+            // Open a Launcher session
             try (LauncherSession session = LauncherFactory.openSession()) {
                 Launcher launcher = session.getLauncher();
 
-                launcher.registerTestExecutionListeners(listener);
+                //Create Listeners
+                SummaryGeneratingListener summaryListener = new SummaryGeneratingListener();
+                OutputTestExecutionListener outputTestExecutionListener = new OutputTestExecutionListener(outputStream);
 
-                TestPlan testPlan = launcher.discover(request);
+                // Register Listeners
+                launcher.registerTestExecutionListeners(summaryListener, outputTestExecutionListener);
 
-                launcher.execute(testPlan);
-            }
+                // Load all build files
+                JUnitContextClassLoader.loadClassesFromDirectory(baseRootPath);
+                List<DiscoverySelector> selectors = new ArrayList<>();
```

Comment:
Currently loads all files from a single given directory (must include test files and dependency files for that). It is unclear how this will perform with external libraries in the tester files.